drivers/I2C.cpp

```diff
@@ -120,6 +124,82 @@ void I2C::unlock() {
     _mutex->unlock();
 }
 
+int I2C::reset() {
+    lock();
+    // Try i2c_reset() i.e. target specific reset
+    i2c_reset(&_i2c);
```

Comment:
This doesn't return pass or fail, that's a bummer. Seems like duplicated efforts.